internal/pkg/agent/application/application.go

```diff
@@ -177,7 +177,8 @@ func New(
 				EndpointSignedComponentModifier(),
 			)
 
-			managed, err = newManagedConfigManager(ctx, log, agentInfo, cfg, store, runtime, fleetInitTimeout, upgrader)
+			// TODO: stop using global state
```

Comment:
Is this TODO referred to the `paths.Top()` in the newManagedConfigManager() call in the next line ?

Reply:
Yes, it is there in the hope we'll manage to remove all this global state at some point in the future.